jrt/src/com/yahoo/jrt/Connection.java

```diff
@@ -89,21 +90,23 @@ private void setState(int state) {
     }
 
     public Connection(TransportThread parent, Supervisor owner,
-                      SocketChannel channel) {
+                      SocketChannel channel, boolean tcpNoDelay) {
 
         this.parent = parent;
         this.owner = owner;
         this.socket = parent.transport().createServerCryptoSocket(channel);
         this.spec = null;
+        this.tcpNoDelay = tcpNoDelay;
         server = true;
         owner.sessionInit(this);
     }
 
-    public Connection(TransportThread parent, Supervisor owner, Spec spec, Object context) {
+    public Connection(TransportThread parent, Supervisor owner, Spec spec, Object context, boolean tcpNoDelay) {
         super(context);
         this.parent = parent;
         this.owner = owner;
         this.spec = spec;
+        this.tcpNoDelay = tcpNoDelay;
```

Comment:
Do we have any numbers that indicate disabling `TCP_NODELAY` is worth it? We unconditionally enabled nodelay for everything to get rid  of spurious latency spikes incurred by the in-kernel deferred packet sending.

Reply:
I saw significantly lower system cpu when hardcoding it yesterday.
However I did a lot of different tests then, and I am not putting more into it than it is a dimension that should be tested further. And for that to happen I need config control over it.